nimare/diagnostics.py

```diff
@@ -139,61 +138,59 @@ def transform(self, result):
         meta_ids = estimator.inputs_["id"]
-        rows = ["Center of Mass"] + list(meta_ids)
-
-        # Let's label the clusters in the thresholded map so we can use it as a NiftiLabelsMasker
-        # This won't work when the Estimator's masker isn't a NiftiMasker... :(
-        conn = ndimage.generate_binary_structure(rank=3, connectivity=1)
-        labeled_cluster_arr, n_clusters = ndimage.label(thresh_arr, conn)
-        labeled_cluster_img = nib.Nifti1Image(
-            labeled_cluster_arr,
-            affine=target_img.affine,
-            header=target_img.header,
+        rows = list(meta_ids)
+
+        # Get clusters table
+        two_sided = (target_img.get_fdata() < 0).any()
+        stat_threshold = self.voxel_thresh or 0
+        clusters_table, label_maps = _get_clusters_table(
+            target_img, stat_threshold, two_sided=two_sided, return_label_maps=True
         )
 
-        if n_clusters == 0:
+        if clusters_table.shape[0] == 0:
             LGR.warning("No clusters found")
-            contribution_table = pd.DataFrame(index=rows)
-            return contribution_table, labeled_cluster_img
-
-        cluster_coms = _get_cluster_coms(labeled_cluster_arr)
-        cluster_coms = vox2mm(cluster_coms, target_img.affine)
-
-        cluster_com_strs = []
-        for i_peak in range(cluster_coms.shape[0]):
-            x, y, z = cluster_coms[i_peak, :].astype(int)
-            xyz_str = f"({x}, {y}, {z})"
-            cluster_com_strs.append(xyz_str)
-
-        # Mask using a labels masker, so that we can easily get the mean value for each cluster
-        cluster_masker = input_data.NiftiLabelsMasker(labeled_cluster_img)
-        cluster_masker.fit(labeled_cluster_img)
-
-        # Create empty contribution table
-        contribution_table = pd.DataFrame(index=rows, columns=list(range(1, n_clusters + 1)))
-        contribution_table.index.name = "Cluster ID"
-        contribution_table.loc["Center of Mass"] = cluster_com_strs
-
-        with tqdm_joblib(tqdm(total=len(meta_ids))):
-            jackknife_results = Parallel(n_jobs=self.n_cores)(
-                delayed(self._transform)(
-                    study_id,
-                    all_ids=meta_ids,
-                    dset=dset,
-                    estimator=estimator,
-                    target_value_map=target_value_map,
-                    stat_values=stat_values,
-                    original_masker=original_masker,
-                    cluster_masker=cluster_masker,
+            contribution_table = pd.DataFrame(columns=["id"], data=rows)
+            return contribution_table, clusters_table, label_maps
+
+        contribution_tables = []
+        signs = [1, -1] if len(label_maps) == 2 else [1]
+        for sign, label_map in zip(signs, label_maps):
+            cluster_ids = sorted(list(np.unique(label_map.get_fdata())[1:]))
+
+            # Create contribution table
+            col_name = "PosTail" if sign == 1 else "NegTail"
```

Comment:
I don't have a great suggestion for what the name should be, maybe it should spell out `PositiveTail`, because it was not immediately clear in the example notebook. 

Regardless, If the fact "PosTail represents positive clusters and NegTail represents negative clusters" is added into the documentation of contribution_table then I would feel good about using whatever names you choose.